build-tools/packages/build-tools/src/repoPolicyCheck/handlers/peerDependencies.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { EOL as newline } from "os";
+import sortPackageJson from "sort-package-json";
+
+import { IPackage } from "../../common/npmPackage";
+import { Handler, readFile, writeFile } from "../common";
+
+export const handler: Handler = {
+    name: "azure-client-peerDependencies",
+    match: /.*?azure-client\/package\.json/i,
+    handler: (file) => {
+        let jsonStr: string;
+        let pkgJson: IPackage;
+        try {
+            jsonStr = readFile(file);
+            pkgJson = JSON.parse(jsonStr);
+        } catch (err) {
+            return "Error parsing JSON file: " + file;
+        }
+
+        if (pkgJson.peerDependencies === undefined) {
+            return `azure-client must have a peerDependencies field in package.json`;
+        }
+
+        if (
+            pkgJson.peerDependencies["fluid-framework"] !==
+            pkgJson.peerDependencies["@fluidframework/core-interfaces"]
```

Comment:
What if core-interfaces is not used as dependency ? Would some sort of “missing client package reference” error be appropriate here ?